### PR TITLE
Remove some inner classes from MasterService

### PR DESCRIPTION
**Pull request by DaveCTurner**
Today the `MasterService` has inner classes `TaskInputs`, `TaskOutputs`
and `BatchExecutionResult` that collect up various data needed for the
cluster state update computation. In fact they're not necessary: we
already have everything we need at the top level except the new
`ClusterState`, which we can return directly.

### CODE REVIEW

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/part-1

File: server/src/main/java/org/elasticsearch/cluster/service/MasterService.java [hunk 1]
```diff
@@ -296,7 +307,92 @@ private void runTasks(TaskInputs taskInputs) {
...
+                            .filter(Objects::nonNull)
+                            .map(
+                                contextPreservingAckListener -> new TaskAckListener(
+                                    contextPreservingAckListener,
+                                    newClusterState.version(),
+                                    newClusterState.nodes(),
+                                    threadPool
+                                )
+                            )
+                            .collect(Collectors.toList())
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/cluster/service/MasterService.java:324]: I believe this was moved from `createAckListener`, however is it possible to update it to use immutable list? 
```suggestion
                            .toList()
```

**Review by idegtiarenko — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/cluster/service/MasterService.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/cluster/service/MasterService.java]: Good point, yes.

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/rest-compatibility

**Pull request merged**